### PR TITLE
[p3a] Guard P3A references with buildflags

### PR DESCRIPTION
**Pull request by freespeech4ever**
Resolves brave/brave-browser#12450

## Submitter Checklist:

- [X] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [X] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by simonhong** [comment]
Welcome and thanks for contributing @freespeech4ever :)
Created https://github.com/brave/brave-core/pull/7009 to run this in CI.

**Review by simonhong — APPROVED** [review]
++

File: browser/brave_browser_process_impl.cc [hunk 1]
```diff
@@ -330,6 +330,7 @@ void BraveBrowserProcessImpl::OnTorEnabledChanged() {
 }
 #endif
 
+if BUILDFLAG(BRAVE_P3A_ENABLED)
```
Comment (simonhong) [inline browser/brave_browser_process_impl.cc:333]: @freespeech4ever   you should use `#if`.

**Review by iefremov — CHANGES_REQUESTED** [review]
We actually don't want to support build wihout P3A. The only reason the corresponding build guard exists is that Android is not ready for it yet. Once we fully support P3A on android most probably we will drop the build flag.

cc @bsclifton

**Comment by bsclifton** [comment]
@iefremov that is correct; we only have the guards because Android isn't ready (yet). However, we will be addressing that soon and removing the guards completely 😄 https://github.com/brave/brave-browser/issues/6176 can be followed for more info

**Pull request closed without merging**